iceoryx_utils/include/iceoryx_utils/cxx/function.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_UTILS_FUNCTION_HPP
+#define IOX_UTILS_FUNCTION_HPP
+
+#include "iceoryx_utils/internal/cxx/static_storage.hpp"
+#include "iceoryx_utils/internal/cxx/storable_function.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+/// @note set the storage type and reorder template arguments for the user API
+/// the storage type must precede the (required) variadic arguments in the internal one
+/// if the static storage is insufficient to store the callable we get a compile time error
+
+/// @note this alias is needed to improve usability
+///       for the API see storable_function
+template <typename Signature, uint64_t Bytes = 128>
```

Comment:
Why not expose the 128 to the user so the can set the magic number?

Reply:
And it needs the `U`

Reply:
It is exposed but defaulted to a reasonable 128.